apps/alert_processor/lib/user_update_worker.ex

```diff
@@ -157,13 +166,18 @@ defmodule AlertProcessor.UserUpdateWorker do
   # Ignore empty updates. The user changed a property we aren't interested in saving locally.
   defp update_user_record(%{updates: updates}) when updates == %{}, do: :ok
 
-  defp update_user_record(%{user_id: user_id, updates: updates}) do
-    case User.get(user_id) do
-      nil ->
+  defp update_user_record(%{id: user_id, mbta_uuid: mbta_uuid, updates: updates} = attribute_map) do
+    user_list = User.get_by_alternate_id(attribute_map |> Map.delete(:updates))
+
+    case length(user_list) do
```

Comment:
suggestion: rather than calculating the length of the list (which is linear in the length of the list) and matching on that, you can match on the list directly:
```elixir
case user_list do
  [] -> # empty case
  [user] -> # single user case
  _ -> # more than 1 user case
end
```

Reply:
💡

Reply:
💡